app/src/main/java/net/kwatts/powtools/adapters/RideListAdapter.java

```diff
@@ -104,4 +117,18 @@ void bind(RideRow rideRow) {
         }
     }
 
+    void loadTrack(long rideId, @NonNull ImageView imageView) {
+        App.dbExecute(database -> {
+            List<Pair<Double, Double>> track = new ArrayList<>();
+            List<Moment> moments = database.momentDao().getFromRide(rideId);
```

Comment:
since you're not using time as a qualifier like the map fragment is, maybe we should add ORDER BY Date ASC just to make sure

Reply:
`getFromRide` is accessed elsewhere, feel free to make the change you suggested if you don't think it would have breaking effects elsewhere.